src/ducks/sep6WithdrawAsset.ts

```diff
@@ -0,0 +1,328 @@
+        networkUrl: networkConfig.url,
+        homeDomain,
+      });
+
+      // Check info
+      const infoData = await checkDepositWithdrawInfo({
+        type: CheckInfoType.WITHDRAWAL,
+        transferServerUrl: tomlResponse.TRANSFER_SERVER,
+        assetCode,
+      });
+
+      const assetInfoData = infoData.withdraw[assetCode];
+
+      const {
+        authentication_required: isAuthenticationRequired,
+      } = assetInfoData;
+
+      let payload = {
+        assetCode,
+        withdrawTypes: { types: { ...assetInfoData.types } },
+        fields: {},
+        kycServer: "",
+        status: ActionStatus.NEEDS_INPUT,
+        token: "",
+        transferServer: tomlResponse.TRANSFER_SERVER,
+      };
+
+      if (isAuthenticationRequired) {
+        // Re-check toml for auth endpoint
+        const webAuthTomlResponse = await checkTomlForFields({
+          sepName: "SEP-6 withdrawal",
+          assetIssuer,
+          requiredKeys: [
+            TomlFields.WEB_AUTH_ENDPOINT,
+            TomlFields.SIGNING_KEY,
+            TomlFields.KYC_SERVER,
+          ],
+          networkUrl: networkConfig.url,
+          homeDomain,
+        });
+        log.instruction({
+          title:
+            "SEP-6 withdrawal is enabled, and requires authentication so we should go through SEP-10",
+        });
+        // SEP-10 start
+        const challengeTransaction = await sep10AuthStart({
+          authEndpoint: webAuthTomlResponse.WEB_AUTH_ENDPOINT,
+          serverSigningKey: webAuthTomlResponse.SIGNING_KEY,
+          publicKey,
+          homeDomain,
+        });
+
+        // SEP-10 sign
+        const signedChallengeTransaction = sep10AuthSign({
+          secretKey,
+          networkPassphrase: networkConfig.network,
+          challengeTransaction,
+        });
+
+        // SEP-10 send
+        const token = await sep10AuthSend({
+          authEndpoint: webAuthTomlResponse.WEB_AUTH_ENDPOINT,
+          signedChallengeTransaction,
+        });
+
+        // Get SEP-12 fields
+        log.instruction({
+          title: "Making GET `/customer` request for user",
+        });
+
+        const sep12Fields = await collectSep12Fields({
+          publicKey,
+          token,
+          kycServer: webAuthTomlResponse.KYC_SERVER,
+        });
+
+        payload = {
+          ...payload,
+          kycServer: webAuthTomlResponse.KYC_SERVER,
+          token,
+        };
+
+        if (sep12Fields) {
+          payload = {
+            ...payload,
+            fields: { ...payload.fields, ...sep12Fields },
+          };
+        }
+      }
+
+      return payload;
+    } catch (error) {
+      const errorMessage = getErrorMessage(error);
+
+      log.error({
+        title: "SEP-6 withdrawal failed",
+        body: errorMessage,
+      });
+
+      return rejectWithValue({
+        errorString: errorMessage,
+      });
+    }
+  },
+);
+
+export const submitSep6DWithdrawFields = createAsyncThunk<
```

Comment:
🤔  This letter "D" looks like a typo to me. Should we rename this function to `submitSep6WithdrawFields`?

Reply:
Ah, good catch - foiled by copy/paste :)